name: django-testsuite
on:
  push:
    branches: [master]
  pull_request:
jobs:
  django-testsuite-3_1_7:
    runs-on: ubuntu-latest
    defaults:
      run:
        working-directory: django
    steps:
      - uses: actions/checkout@v2
        with:
          path: ddtrace
      - uses: actions/checkout@v2
        with:
          repository: django/django
          ref: 3.1.7
          path: django
      - uses: actions/setup-python@v2
        with:
          python-version: "3.8"
      - name: Install pylibmc's libmemcached package
        # Django-specific: pylibmc in Ubuntu requires libmemcached package
        run: sudo apt install libmemcached-dev zlib1g
      - name: Install dependencies
        # Django-specific: separate dependencies for tests
        run: pip install -r tests/requirements/py3.txt
      - name: Install ddtrace
        run: pip install ../ddtrace
      - name: Install django
        run: pip install ../django
      - name: Run tests
        # Django's test directory contains a 'requests' test module
        # This interferes with the requests library patching in the tracer
        # test_supports_json_field_operational_error will fail with the tracer
        run: DD_TRACE_REQUESTS_ENABLED=0 ddtrace-run tests/runtests.py -k "not backends.sqlite.test_features.FeaturesTests.test_supports_json_field_operational_error"


